staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -702,7 +702,8 @@ func validateCustomResourceDefinitionValidation(customResourceValidation *apiext
 			requireValidPropertyType: opts.requireValidPropertyType,
 		}
 
-		allErrs = append(allErrs, ValidateCustomResourceDefinitionOpenAPISchema(schema, fldPath.Child("openAPIV3Schema"), openAPIV3Schema, true, &opts)...)
+		schemaIsNonStructural := schemaIsNonStructural(schema)
```

Comment:
how does this relate to `requireStructuralSchema` below? Cant we just use `requireStructuralSchema` instead?

Reply:
Updated to use `requireStructuralSchema` to enforce XValidations only in structural schema.
Please let me know if it needs more validation.